js/components/utils.js

```diff
@@ -432,37 +432,78 @@ var Utils = {
    * @function arrayDeepMerge
    * @param {Array} target - An array that will receive new items if additional items are passed
    * @param {Array} source - An array containing additional items to merge into target
-   * @param {Object} optionsArgument - optional parameters passed, i.e. arrayMerge function
+   * @param {Object} optionsArgument - optional parameters passed, i.e. arrayMerge, swap, unionBy, clone, arrayFusion
    * @returns {Array} new merged array with items from both target and source
    */
   arrayDeepMerge: function(target, source, optionsArgument) {
+    // replaces target with source, no merge
+    if (optionsArgument.arrayFusion === 'replace') {
```

Comment:
I added more info to the description of this PR that links to deep merge lib and how this function is used.

We don't call this function directly, it's passed as a callback using `arrayMerge` param in `optionsArgument` object. When this function is called `optionsArgument` will always at least have `arrayMerge`.

Will update doc language to remove "optional" for this param